### ISSUE TITLE
Sensor not showing up correctly on current (0.14-dev) beta site

### ISSUE THREAD

**Issue by bsimmo** [ConnectedHumber/Air-Quality-Web#66]
An introduction of a new sensor reading type gtemp for Ground Temperature.
and the introduction of new reading types for DEFRA sensors
It is also noticed that a new Pi setup is missing it's PM.

MQTT section have been updated
Database has been updated with details
by @BNNorman 
See
https://github.com/ConnectedHumber/MQTT

On the current live site v0.13.6 this sensor readings are displaying correctly.

Unfortunately, the current 0.14-dev site is missing these readings.
This can also be seen using the API

https://sensors.connectedhumber.org/api.php?action=list-reading-types&device-id=61
https://sensors.connectedhumber.org/beta/api.php?action=list-reading-types&device-id=61

https://sensors.connectedhumber.org/api.php?action=list-reading-types&device-id=59
https://sensors.connectedhumber.org/beta/api.php?action=list-reading-types&device-id=59
They are being shown under the Info tab (e.g. this 'beta' display for #59 above)
![image](https://user-images.githubusercontent.com/11537335/109681131-134aee80-7b75-11eb-9cf9-3112a9509c2a.png)
 

The new DEFRA Sensor seem to be working correctly (I have only checked #60 and #61, the newly added DEFRA locations)

Talk has been in Mattermost see approximately https://mattermost.connectedhumber.org/default/pl/mdcaweo9tjfupqjdcpyua6esdc

But moving to github as it should be easier to follow.

Assumption is it's something to do with the new

I'll let @sbrl and @BNNorman who know more add more details if they know any.

**Comment by sbrl** [comment]
Ok, so I've taken a look, and here's an example of the data your sensor sends in:

```json
{'dev': 'CHGSN-5664433-1', 'gtemp': '1.06', 'temp': '2.7', 'pressure': '1009.2', 'humidity': '71.4', 'timestamp': '2021-02-8T13:49:17+00:00'}
```

This specifies following reading types:
```
gtemp
temp
pressure
humidity
```

According to the database, your device is registered with the following sensors:

```
MariaDB [aq_db]> SELECT * FROM sensors WHERE sensors.id = 1 OR sensors.id = 8;
+----+---------+-------------------------------+
| id | Type    | Description                   |
+----+---------+-------------------------------+
|  1 | BME280  | Temperature,Pressure,Humidity |
|  8 | DS18B20 | Temperature                   |
+----+---------+-------------------------------+
```

The `BME280` sensor is registered in the database to provide the following reading types:

```
| humidity      | Humidity (percent)         |  1 |
| pressure      | Pressure (mBar)            |  4 |
| temperature   | Temperature (C)            |  5 |
```

The `DS18B20` is registered to provide the following reading types:

```
| temperature | Temperature (C) |  5 |
```

The problem here is that the entry in the database for the `DS18B20` is incorrect. It should not be registered to provide `temperature` - the `BM280` is already registered for that.

It should instead be registered to provide `gtemp`. The following SQL should fix the problem, though it is untested:

```
DELETE FROM sensor_reading_value_types WHERE id = 16;
INSERT INTO sensor_reading_value_types (sensor_id, reading_value_types_id) VALUES (8, 17);
```

**Comment by bsimmo** [comment]
This leaves a problem in the future
DS18B20 does not always read gtemp.
Somebody could very easily be using it in place of a BME280 and sending it in a temperature. 
It is a Sensor of type Temperature reporting Ground Temperature readings in degC as many other sensors are, and may be used together. 
Temperature should be a key, if it is, just a description. 
What are we going to do if a device has three DS18B20 Ground Temperature sensor coming in as gtemp#
It is a possible use case as i can cover three different types of ground. 

Grass, concrete Pavement, stone, shaded etc. 

I understand the fix to get it working, but is there a better solution in the long run? 


Who knows what projects @adamjennison will come up with.

**Comment by BNNorman** [comment]
For defra sensors i created defra_pm etc. Creating ds18b20_gnd etc should solve the problem. We could create a new name for each use case.
Otherwise the database schema needs changing, carefully.

**Comment by BNNorman** [comment]
I have added a new sensor DS18B20_GND -> gtemp
I have removed the DS18B20 from ben's device and added the DS18B20_GND sensor.

If there is a need to have multiple ground sensors on one device I suggest we name them DS18B20_GND1,DS18B20_GND2 etc

The only other solution would be to add another 'useage' table which will break the SQL and change the schema. Such a solution would be  sensor->usage->reading_value_type. It would be a bit of a pain because all current sensors would need to be added/changed/whatever. 

Using different sensor names , as suggested, doesn't break anything and is self explanatory BUT may need additional reading_value_types to go with them like gtemp1, gtemp2. dbLoader.py might not need changing because I coded it to load the reading_value_types into a dictionary at startup only. Additions to the reading_value_types table would require us to restart the dbLoader (systemctl restart dbLoader)
